runtime/doc/quickfix.txt

```diff
@@ -43,12 +43,27 @@ From inside Vim an easy way to run a command and handle the output is with the
 The 'errorformat' option should be set to match the error messages from your
 compiler (see |errorformat| below).
 
-							*quickfix-ID*
+					*quickfix-stack* *quickfix-ID*
 Each quickfix list has a unique identifier called the quickfix ID and this
 number will not change within a Vim session. The |getqflist()| function can be
 used to get the identifier assigned to a list. There is also a quickfix list
-number which may change whenever more than ten lists are added to a quickfix
-stack.
+number which may change whenever more than 'chistory' lists are added to a
+quickfix stack.
+
+					*quickfix-list-allocation* *E1520*
+Note that when allocating or resizing a location or quickfix list stack, if a
+memory error occurs, it will try allocating one less than the given size,
+until there is a successful allocation. If allocation still fails when just
+allocating enough memory for a single quickfix list or the original
+maximum stack size (when resizing), an error is returned. When shrinking a
+quickfix/location list stack, please note that if it fails, then any lists
+that would be removed in order to fit the new stack size, will still be
+removed.
```

Comment:
Ditto.

```suggestion
Note that when allocating or resizing a location or quickfix list stack, if a
memory error occurs, it will try allocating one less than the given size,
until there is a successful allocation.  If allocation still fails when just
allocating enough memory for a single quickfix list or the original maximum
stack size (when resizing), an error is returned.  When shrinking a
quickfix/location list stack, please note that if it fails, then any lists
that would be removed in order to fit the new stack size, will still be
removed.
```